securedrop/tests/test_journalist.py

```diff
@@ -809,6 +809,40 @@ def test_user_resets_user_hotp_format_non_hexa(journalist_app, test_journo):
     assert old_secret == user.otp_secret
 
 
+def test_user_resets_user_hotp_error(mocker,
+                                     journalist_app,
+                                     test_journo):
+    bad_secret = '1234'
+    old_secret = test_journo['otp_secret']
+    error_message = 'SOMETHING WRONG!'
+    mocked_error_logger = mocker.patch('journalist.app.logger.error')
+
+    with journalist_app.test_client() as app:
+        _login_user(app, test_journo['username'], test_journo['password'],
+                    test_journo['otp_secret'])
+
+        mocker.patch('models.Journalist.set_hotp_secret',
+                     side_effect=TypeError(error_message))
+
+        with InstrumentedApp(journalist_app) as ins:
+            app.post('/account/reset-2fa-hotp',
+                     data=dict(uid=test_journo['id'],
```

Comment:
Nit: `uid=test_journo['id']` not necessary here (since it's `/account/reset-2fa-hotp` not `/admin/reset-2fa-hotp`)